core/src/stored/backends/chunked_device.h

```diff
@@ -28,11 +28,18 @@
 #ifndef BAREOS_STORED_BACKENDS_CHUNKED_DEVICE_H_
 #define BAREOS_STORED_BACKENDS_CHUNKED_DEVICE_H_
 
+#include <sys/types.h>
+#include "stored/dev.h"
+#include "ordered_cbuf.h"
+#include <optional>
+
 template <typename T> class alist;
 
-#include "ordered_cbuf.h"
 namespace storagedaemon {
 
+class DeviceControlRecord;
+struct DeviceStatusInformation;
```

Comment:
Shouldnt we include headers for definitions like these ?

If we cannot do this without linking to libbareos, we should think about (read: create tasks for) creating headers that only export the stuff that backends actually need.

Reply:
I think we once settled on forward declarations in those cases, but that doesn't mean we shouldn't have these in header files.